notation.go

```diff
@@ -63,31 +45,202 @@ type SignOptions struct {
 // and packing in various signature formats.
 type Signer interface {
 	// Sign signs the artifact described by its descriptor,
-	// and returns the signature.
-	Sign(ctx context.Context, desc Descriptor, opts SignOptions) ([]byte, error)
+	// and returns the signature and SignerInfo.
+	Sign(ctx context.Context, desc ocispec.Descriptor, envelopeMediaType string, opts SignOptions) ([]byte, *signature.SignerInfo, error)
+}
+
+// Sign signs the artifact in the remote registry and push the signature to the
+// remote.
+// The descriptor of the sign content is returned upon sucessful signing.
+func Sign(ctx context.Context, signer Signer, repo registry.Repository, opts SignOptions) (ocispec.Descriptor, error) {
+	targetDesc, err := repo.Resolve(ctx, opts.ArtifactReference)
+	if err != nil {
+		return ocispec.Descriptor{}, err
+	}
+	sig, signerInfo, err := signer.Sign(ctx, targetDesc, opts.SignatureMediaType, opts)
+	if err != nil {
+		return ocispec.Descriptor{}, err
+	}
+	annotations, err := generateAnnotations(signerInfo)
+	if err != nil {
+		return ocispec.Descriptor{}, err
+	}
+	_, _, err = repo.PushSignature(ctx, opts.SignatureMediaType, sig, targetDesc, annotations)
+	if err != nil {
+		return ocispec.Descriptor{}, err
+	}
+
+	return targetDesc, nil
 }
 
 // VerifyOptions contains parameters for Verifier.Verify.
 type VerifyOptions struct {
+	// ArtifactReference is the reference of the artifact that is been
+	// verified against to.
+	ArtifactReference string
+
 	// SignatureMediaType is the envelope type of the signature.
-	// Currently both `application/jose+json` and `application/cose` are supported.
+	// Currently both `application/jose+json` and `application/cose` are
+	// supported.
 	SignatureMediaType string
+
+	// PluginConfig is a map of plugin configs.
+	PluginConfig map[string]string
+
+	// MaxVerificationLimit is the maximum number of signature envelopes that
+	// can be associated with the target artifact. If not set by user, it will
+	// be set to 50 by default.
+	// Note: this option is scoped to notation.Verify(). verifier.Verify() is
+	// for signle signature verification, and therefore, does not use it.
+	MaxVerificationLimit int
```

Comment:
Will it be more accurate to say
```suggestion
	MaxSignatureAttempts int
```